docs/contributing/semgrep-contributing.md

```diff
@@ -2,50 +2,84 @@
 
-## Getting `semgrep-core` binary
+## Setting up the environment
 
-If you would like to install `semgrep-core` from source (for example, because you want to fix a parse error), follow the instructions in [Building `semgrep-core`](semgrep-core-contributing.md#building-semgrep-core) and skip this section.
+You will need Python >= 3.6.
 
-Otherwise, visit the [releases page](https://github.com/returntocorp/semgrep/releases)
-and grab the latest zipfile or tarball for your platform. Extract this archive
-and inside should be the necessary binaries. You can confirm this by running:
+Most Python development is done inside the `semgrep` directory (from the top level of this repo, `semgrep/semgrep/`):
 
 ```bash
-$ ./semgrep-core --help
+$ cd semgrep
 ```
 
-Copy this file to somewhere in your `$PATH` so `semgrep` can find them. For
-example, you may create a `~/bin/` directory within the repository. [Include it in your `$PATH`](https://unix.stackexchange.com/questions/26047/how-to-correctly-add-a-path-to-path)
-and run the binary from there.
+We use [`pipenv`](https://github.com/pypa/pipenv) to manage our virtual environment.
+You can install it like this:
 
-Alternatively, you may include it somewhere like `/usr/local/bin/`.
+```bash
+$ python -m pip install pipenv
+```
 
-## Setting up the environment
+Next we need to initialize the environment.
+This command will install dev dependencies such as pytest and will also install semgrep in editable mode in the pipenv.
 
-Once you have `semgrep-core` installed, you will be able to build `semgrep`. You will need Python >= 3.6 as well.
+```bash
+$ SEMGREP_SKIP_BIN=true python -m pipenv install --dev
+```
 
-The `semgrep` command is a Python wrapper around the `semgrep-core` executable which is written in OCaml. Most Python development is done inside the `semgrep` directory (from the top level of this repo, `semgrep/semgrep/`):
+:::note
+SEMGREP_SKIP_BIN` tells the installer that we will bring our own semgrep-core; see below.*
+:::
 
-```
-$ cd semgrep
-```
+## Getting the `semgrep-core` binary
 
-We use [`pipenv`](https://github.com/pypa/pipenv) to manage our virtual
-environment. If you don't have `pipenv` installed, the following command will do
-so for you:
+Almost all usages of `semgrep` require the `semgrep-core` binary.
+To get this binary,
+your safest bet is to follow the instructions in [Building `semgrep-core`](semgrep-core-contributing.md#building-semgrep-core),
+which takes around 20 minutes.
 
-```
-$ python -m pip install pipenv
-```
+Two shortcuts are available as alternatives,
+where you use a pre-compiled binary.
+The downsides of using a pre-compiled binary are:
+
+1. You will not be able to make edits to `semgrep-core`,
+   for example to fix a parse error.
+2. Semgrep will fail if the interface between `semgrep` and `semgrep-core` has changed
+   since the binary was compiled.
+   This has historically been happening around every two months.
```

Comment:
@emjin did I guess this frequency right?

Reply:
I would just say that it happens often.

Reply:
I'd like to be more specific if possible 😕 Some could interpret that as daily, some as monthly.